src/wp-includes/media.php

```diff
@@ -1002,74 +1003,26 @@ function wp_get_attachment_image_sizes( $attachment_id, $size = 'medium', $args
 		return false;
 	}
 
-	// Set the image width in pixels.
-	$img_width = $img_width . 'px';
-
 	// Set up our default values.
-	$defaults = array(
-		'sizes' => array(
-			array(
-				'size_value' => '100vw',
-				'mq_value'   => $img_width,
-				'mq_name'    => 'max-width'
-			),
-			array(
-				'size_value' => $img_width
-			),
-		)
+	$sizes = array(
+		'(max-width: ' . $img_width . 'px) 100vw',
+		$img_width . 'px',
 	);
 
-	$args = wp_parse_args( $args, $defaults );
-
 	/**
-	* Filter arguments used to create 'sizes' attribute.
+	* Filter the values used to create a 'sizes' attribute.
 	*
 	* @since 4.4.0
 	*
-	* @param array   $args          An array of arguments used to create a 'sizes' attribute.
-	* @param int     $attachment_id Post ID of the original image.
-	* @param string  $size          Name of the image size being used.
+	* @param array        $sizes         An array of values used to create a 'sizes' attribute.
+	* @param int          $attachment_id Post ID of the original image.
+	* @param string|array $size          Image size name or a flat array of width and height values
+	*                                    of the image being used.
 	*/
-	$args = apply_filters( 'wp_image_sizes_args', $args, $attachment_id, $size );
-
-	// If sizes is passed as a string, just use the string.
-	if ( is_string( $args['sizes'] ) ) {
-		$size_list = $args['sizes'];
-
-	// Otherwise, breakdown the array and build a sizes string.
-	} elseif ( is_array( $args['sizes'] ) ) {
-
-		$size_list = '';
-
-		foreach ( $args['sizes'] as $size ) {
-
-			// Use 100vw as the size value unless something else is specified.
-			$size_value = ( $size['size_value'] ) ? $size['size_value'] : '100vw';
-
-			// If a media length is specified, build the media query.
-			if ( ! empty( $size['mq_value'] ) ) {
-
-				$media_length = $size['mq_value'];
-
-				// Use max-width as the media condition unless min-width is specified.
-				$media_condition = ( ! empty( $size['mq_name'] ) ) ? $size['mq_name'] : 'max-width';
-
-				// If a media_length was set, create the media query.
-				$media_query = '(' . $media_condition . ": " . $media_length . ') ';
-
-			} else {
-				// If no media length was set, $media_query is blank.
-				$media_query = '';
-			}
-
-			// Add to the source size list string.
-			$size_list .= $media_query . $size_value . ', ';
-		}
-
-		// Remove the trailing comma and space from the end of the string.
-		$size_list = substr( $size_list, 0, -2 );
-	}
+	$sizes = apply_filters( 'wp_image_sizes_values', $sizes, $attachment_id, $size );
```

Comment:
We should pass `$img_width` to the filter so anyone hooking in doesn't have to duplicate the call to `image_get_intermediate_size()`.